packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1625,16 +1625,21 @@ export class StripeHelper extends StripeHelperBase {
     );
   }
 
-  getPaymentProvider(customer: Stripe.Customer) {
-    const subscription = customer.subscriptions?.data.find((sub) =>
-      ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status)
-    );
-    if (subscription) {
-      return subscription.collection_method === 'send_invoice'
-        ? 'paypal'
-        : 'stripe';
+  getPaymentProvider(
+    customer: Stripe.Customer
+  ): 'not_chosen' | 'paypal' | 'stripe' {
+    const defaultPaymentMethod: string | Stripe.PaymentMethod | null =
+      customer.invoice_settings.default_payment_method;
+    const paymentMethodId: string | undefined =
+      customer.metadata.paypalAgreementId;
```

Comment:
Is it necessary to explicitly define the types here?

Where possible, it's preferable for TypeScript to infer the typings, and I believe in this case it should be able to do that in all these cases. Let me know if that's not the case however.